### PR TITLE
protect from moved blueocean api

### PR DESCRIPTION
**Pull request by gabemontero**
Fixes https://github.com/openshift/jenkins-sync-plugin/issues/191

@openshift/sig-developer-experience fyi

### CODE REVIEW

**Comment by gabemontero** [comment]
[test]

File: src/main/java/io/fabric8/jenkins/openshiftsync/BuildSyncRunListener.java [hunk 1]
```diff
@@ -208,14 +208,26 @@ protected synchronized void pollRun(Run run) {
...
+	                break;
+	            }
+	            
+	        }
+		} catch (Throwable t) {
+		    // use of info is intentional ... in case the blue ocean deps get bumped
+		    // by another dependency vs. our bumping it explicitly, I want to
+		    // find out quickly that we need to switch to the new method noted 
+		    // above
+		    logger.log(Level.INFO, "pollRun", t);
```
Comment (bparees) [inline src/main/java/io/fabric8/jenkins/openshiftsync/BuildSyncRunListener.java]: severe or warning instead then?

Comment (gabemontero) [reply src/main/java/io/fabric8/jenkins/openshiftsync/BuildSyncRunListener.java]: i waffled on that ... i'm convinced - switching

**Comment by gabemontero** [comment]
[merge]

**Comment by gabemontero** [comment]
[merge]

**Comment by jim-minter** [comment]
@gabemontero @bparees could/should we be doing more testing-wise here?

**Comment by gabemontero** [comment]
hey @jim-minter , can you clarify a bit?

this particular change in isolation is safe and sufficiently tested IMO

but there are several discussion threads we could go beyond that

based on how you clarify, an actual voice conversation may be warranted, but let's wait and see

**Comment by bparees** [comment]
I assume he's suggesting that we should have a test so that next time this gets broken, we know immediately when we update the blueocean plugin in the image.

**Comment by jim-minter** [comment]
Right - I'm wondering what we need to get ahead of bug reports like #191.

**Comment by gabemontero** [comment]
OK thanks for the clarification ... so, the test "to get ahead" would be:
1) a new image extension, where we upgrade a volatile plugin like blueocean to a version that is later then what we ship in the jenkins image, as later versions come up, but we don't yet choose to 
2) run that image through the pipeline/plugin extended tests to validate sync and/or client plugin functionality

where with today's structure, those tests are located in openshift/origin

In my mind at least, boils down to the "external repo" test conundrum, and certainly can be another item on our burgeoning test road map / wish list

But not the type of test we can prereq changes for at this time.  In the past, we've added such tests during the release cycle, and by the time the rhel image is out, it is automated.

Given the nature of the error (no method) and the isolated access to blueocean in sync plugin, the lack of risk of the change, along with the eagerness of the redhatter onsite, simply merged after our current regression passing.

**Comment by bparees** [comment]
> a new image extension, where we upgrade a volatile plugin like blueocean to a version that is later then what we ship in the jenkins image, as later versions come up, but we don't yet choose to

not sure why we would need to do this.  When we update the plugins.txt in such a way that the new blueocean plugin version is picked up, the PR should fail the test we introduce.  

I would think we just need to introduce a test that hits this codepath.  Are you saying our current pipeline/plugin extended tests already hit this codepath?  If so, why aren't PRs that updated the blueocean plugin version failing?

Or was this a case of someone taking our image and updating the blueocean plugin themselves?

**Comment by bparees** [comment]
(if our existing tests are already catching this failure and users only hit it when they manually switch to a level of blueocean that we didn't distribute, i don't think that's something we need to introduce new tests for and the log warning is probably sufficient).

**Comment by gabemontero** [comment]
On Thu, Jan 18, 2018 at 1:29 PM, Ben Parees <notifications@github.com>
wrote:

> a new image extension, where we upgrade a volatile plugin like blueocean
> to a version that is later then what we ship in the jenkins image, as later
> versions come up, but we don't yet choose to
>
> not sure why we would need to do this. When we update the plugins.txt in
> such a way that the new blueocean plugin version is picked up, the PR
> should fail the test we introduce.
>
this is what I meant by "image extension" ... perhaps an overloaded term
... perhaps more accurate "we need to generate an image where the standard
plugin list is modifed"
see clarification at bottom

> I would think we just need to introduce a test that hits this codepath.
> Are you saying our current pipeline/plugin extended tests already hit this
> codepath? If so, why aren't PRs that updated the blueocean plugin version
> failing?
>
no/no

> Or was this a case of someone taking our image and updating the blueocean
> plugin themselves?
>
yes that is what they did

**Comment by gabemontero** [comment]
On Thu, Jan 18, 2018 at 1:31 PM, Ben Parees <notifications@github.com>
wrote:

> (if our existing tests are already catching this failure and users only
> hit it when they manually switch to a level of blueocean that we didn't
> distribute, i don't think that's something we need to introduce new tests
> for and the log warning is probably sufficient).
>
the failure does not occur unless they manually switch the blueocean level
to a level higher than the one we currently distribute in our image

**Comment by bparees** [comment]
i'm still unclear on one thing.  If i created a PR that bumped blue ocean to a newer level (and assuming that newer level changed this api again), would the PR fail the current set of tests?

**Comment by gabemontero** [comment]
To clarify, when I bump the version of a plugin in the base-plugins.txt file in the https://github.com/openshift/jenkins repo, I take the resulting image through our pipeline extended tests.

**Comment by bparees** [comment]
got it. 

So yeah, then I think the overall answer to @jim-minter's request is "in general we need to run the origin extended tests against the jenkins repo PRs" (as you implied and I missed) and we should be fine.

**Pull request merged**